File: app/src/main/java/com/example/zaloraapp/RegisterActivity.java
package com.example.zaloraapp;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.util.Patterns;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

public class RegisterActivity extends AppCompatActivity {

    private EditText etFirstName, etLastName, etEmail, etPassword, etConfirmPassword;
    private Button btnRegister;
    private TextView tvLoginLink;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_register);

        etFirstName       = findViewById(R.id.etFirstName);
        etLastName        = findViewById(R.id.etLastName);
        etEmail           = findViewById(R.id.etEmail);
        etPassword        = findViewById(R.id.etPassword);
        etConfirmPassword = findViewById(R.id.etConfirmPassword);
        btnRegister       = findViewById(R.id.btnRegister);
        tvLoginLink       = findViewById(R.id.tvLoginLink);

        // Register button
        btnRegister.setOnClickListener(v -> {
            String firstName = etFirstName.getText().toString().trim();
            String lastName  = etLastName.getText().toString().trim();
            String email     = etEmail.getText().toString().trim();
            String password  = etPassword.getText().toString().trim();
            String confirm   = etConfirmPassword.getText().toString().trim();

            // Validation
            if (TextUtils.isEmpty(firstName)) {
                etFirstName.setError("First name is required");
                return;
            }
            if (TextUtils.isEmpty(lastName)) {
                etLastName.setError("Last name is required");
                return;
            }
            if (TextUtils.isEmpty(email)) {
                etEmail.setError("Email is required");
                return;
            }
            if (!Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
                etEmail.setError("Enter a valid email address");
                return;
            }
            if (TextUtils.isEmpty(password)) {
                etPassword.setError("Password is required");
                return;
            }
            if (password.length() < 6) {
                etPassword.setError("Password must be at least 6 characters");
                return;
            }
            if (!password.equals(confirm)) {
                etConfirmPassword.setError("Passwords do not match");
                return;
            }

            // TODO: Replace with real registration logic (e.g. Firebase, API call)
            Toast.makeText(this, "Account created successfully!", Toast.LENGTH_SHORT).show();

            // Go to Login after successful registration
            Intent intent = new Intent(RegisterActivity.this, LoginActivity.class);
            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
            startActivity(intent);
        });

        // "Already have an account? Login" link
        tvLoginLink.setOnClickListener(v -> {
            finish(); // go back to LoginActivity
        });
    }
}

File: app/src/main/java/com/example/zaloraapp/RegisterActivity.kt
package com.example.zaloraapp

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class RegisterActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register)

        findViewById<ImageView>(R.id.btnClose).setOnClickListener {
            finish()
        }

        findViewById<Button>(R.id.btnCreateAccount).setOnClickListener {
            // For now, just navigate to Dashboard
            startActivity(Intent(this, DashboardActivity::class.java))
            finish()
        }

        findViewById<TextView>(R.id.tvLogin).setOnClickListener {
            startActivity(Intent(this, LoginActivity::class.java))
            finish()
        }

        // Gender selection logic placeholder
        findViewById<Button>(R.id.btnWomen).setOnClickListener {
            // Update UI state
        }
        findViewById<Button>(R.id.btnMen).setOnClickListener {
            // Update UI state
        }
    }
}
